engine/runtime/action-processor/src/attribute/bulk_renamer.rs

```diff
@@ -0,0 +1,243 @@
+                if let Some(attrs) = &self.params.selected_attributes {
+                    Ok(attrs.iter().map(Attribute::new).collect())
+                } else {
+                    Err(AttributeProcessorError::BulkRenamer(
+                        "No attributes selected for renaming".to_string(),
+                    )
+                    .into())
+                }
+            }
+        }
+    }
+
+    fn rename_attributes(
+        &self,
+        feature: &mut Feature,
+        attributes: Vec<Attribute>,
+    ) -> Result<Vec<Attribute>, BoxedError> {
+        let mut attributes_to_remove = vec![];
+
+        for attr in attributes {
+            if let Some(value) = feature.attributes.get(&attr) {
+                let new_name = self.get_new_name(&attr.inner())?;
+                if new_name.is_empty() {
+                    feature.attributes.remove(&attr);
+                } else {
+                    let new_attr = Attribute::new(new_name);
+                    feature.attributes.insert(new_attr, value.clone());
+                    attributes_to_remove.push(attr.clone());
+                }
+            }
+        }
+        Ok(attributes_to_remove)
+    }
+
+    fn get_new_name(&self, attr_name: &str) -> Result<String, BoxedError> {
+        match self.params.rename_action {
+            RenameAction::AddPrefix => Ok(format!("{}{}", self.params.rename_value, attr_name)),
+            RenameAction::AddSuffix => Ok(format!("{}{}", attr_name, self.params.rename_value)),
+            RenameAction::RemovePrefix => self.remove_prefix(attr_name),
+            RenameAction::RemoveSuffix => self.remove_suffix(attr_name),
+            RenameAction::StringReplace => self.string_replace(attr_name),
+        }
+    }
+
+    fn remove_prefix(&self, attr_name: &str) -> Result<String, BoxedError> {
+        if attr_name.starts_with(&self.params.rename_value) {
+            Ok(attr_name
+                .strip_prefix(&self.params.rename_value)
+                .unwrap_or(attr_name)
+                .to_string())
+        } else {
+            Err(AttributeProcessorError::BulkRenamer(format!(
+                "Attribute '{}' does not start with prefix '{}'",
+                attr_name, self.params.rename_value
+            ))
+            .into())
+        }
+    }
+
+    fn remove_suffix(&self, attr_name: &str) -> Result<String, BoxedError> {
+        if attr_name.ends_with(&self.params.rename_value) {
+            Ok(attr_name
+                .strip_suffix(&self.params.rename_value)
+                .unwrap_or(attr_name)
+                .to_string())
+        } else {
+            Err(AttributeProcessorError::BulkRenamer(format!(
+                "Attribute '{}' does not end with suffix '{}'",
+                attr_name, self.params.rename_value
+            ))
+            .into())
+        }
+    }
+
+    fn string_replace(&self, attr_name: &str) -> Result<String, BoxedError> {
+        if let Some(ref find) = self.params.text_to_find {
```

Comment:
![must](https://img.shields.io/badge/review-must-critical.svg)
Regular expressions should be compiled in advance using the build method.

Reply:
Fixed here https://github.com/reearth/reearth-flow/pull/583/commits/fef6798bec637fdb35e24df76a9072257727df2a